src/rust/graph-descriptions/src/node.rs

```diff
@@ -850,4 +858,80 @@ impl NodeT for Node {
             }
         }
     }
+
+    fn attach_predicates_to_mutation_unit(&self, mutation_unit: &mut MutationUnit) {
```

Comment:
My only concern with this PR is the overhead of 2k new lines that are mostly boilerplate.  Would you be OK with letting me know when this PR is out of draft, and I can write up some macros that would lower the LOC?

Reply:
I would absolutely love that! Thank you for the help. A derive macro is almost certainly a must for this pattern to continue going forward.

Reply:
FWIW once this PR is in I'll be putting another one up that cuts out tons of the code and adds multiple integration tests.